jupyter_client/kernelspec.py

```diff
@@ -6,12 +6,13 @@
 import io
 import json
 import os
+import re
```

Comment:
Indeed! My first implementation was testing `url_quote(name) == name`, but I figured a simple character whitelist would be easier to understand. I think my regex-fu is strong enough for that, at least.